src/modules/emote_menu/components/VirtualizedList.jsx

```diff
@@ -0,0 +1,76 @@
+import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react';
+
+const style = (index, rowHeight) => ({
+  position: 'absolute',
+  top: `${index * rowHeight}px`,
+  height: `${rowHeight}px`,
+  width: '100%',
+});
+
+const headerStyle = (rowHeight) => ({
+  position: 'sticky',
+  top: '0px',
+  height: `${rowHeight}px`,
+  width: '100%',
+});
+
+function VirtualizedList(
+  {totalRows, rowHeight, renderRow, windowHeight, stickyRows = [], onHeaderChange = () => {}, ...restProps},
+  ref = null
+) {
+  const listHeight = useMemo(() => rowHeight * totalRows, [totalRows, rowHeight]);
+  const headers = useMemo(() => stickyRows.sort((a, b) => b - a), [stickyRows]);
```

Comment:
curious: if the rows are build in the same direction as settings, wouldn't the sort be identical?

Reply:
This is included for reusability sake, i.e if the `stickyRows` were ever explicitly defined unordered but I get why you would see this as being redundant.